src/ampycloud/utils/utils.py

```diff
@@ -10,12 +10,168 @@
+    ::
+
+        'ceilo'/pd.StringDtype(), 'dt'/float, 'alt'/float, 'type'/int
+
+    The ``ceilo`` column contains the names/ids of the ceilometers as ``pd.StringDtype()``.
+    See `the pandas documentation <https://pandas.pydata.org/docs/reference/api/pandas.StringDtype.html>`__
+    for more info about this type.
+
+    The ``dt`` column contains time deltas, in seconds, between a given ceilometer
+    observation and ``ref_dt`` (i.e. ``obs_time-ref_dt``). Ideally, ``ref_dt`` would be the issuing
+    time of the METAR message, such that ``dt`` values are negative, with the smallest one
+    corresponding to the oldest measurement.
+
+    The ``alt`` column contains the cloud base hit altitudes reported by the ceilometers, in ft
+    above ground.
+
+    The ``type`` column contains integers that correspond to the hit *sequence id*. If a given
+    ceilometer is reporting multiple hits for a given timestep (corresponding to a cloud level 1,
+    cloud level 2, cloud level 3, etc ...), the ``type`` of these measurements would be ``1``,
+    ``2``, ``3``, etc ... Any data point with a ``type`` of ``-1`` will be flagged in the ampycloud
+    plots as a vertical Visibility (VV) hits, **but it will not be treated any differently than any
+    other regular hit**. Type ``0`` corresponds to no (cloud) detection, in which case the
+    corresponding hit altitude should be a NaN.
+
+    Important:
+        A **non-detection** corresponds to a valid measurement with a ``dt`` value, a ``type 0``,
+        and ``NaN`` as the altitude. It should not be confused with a **non-observation**,
+        when no data was acquired at all !
+
+    If it all sounds confusing, it is possible to obtain an example of the required data format
+    from the :py:func:`.utils.mocker.canonical_demo_data` routine of the package, like so::
+
+        from ampycloud.utils import mocker
+        mock_data = mocker.canonical_demo_data()
+
+    As mentionned above, it is also possible to verify if a given :py:class:`pandas.DataFrame` is
+    meeting the ampycloud requirements ahead of time via
+    :py:func:`ampycloud.utils.utils.check_data_consistency`::
+
+        from ampycloud.utils.utils import check_data_consistency
+        checked_pdf = check_data_consistency(pdf)
+
+    This will raise an :py:class:`ampycloud.errors.AmpycloudError` if:
+
+        * ``pdf`` is not a :py:class:`pandas.DataFrame`.
+        * ``pdf`` is missing a required column.
+        * ``pdf`` has a length of 0.
+
+    In addition, this will raise an :py:class:`ampycloud.errors.AmpycloudWarning` if:
+
+        * any of ``pdf`` column type is not as expected. Note that in this case, the code will try
+          to correct the type on the fly.
+        * ``pdf`` has any superfluous columns. In this case, the code will drop them automatically.
+        * Any hit altitude is negative.
+        * Any ``type 0`` hit has a non-NaN altitude.
+        * Any ``type 1`` hit has a NaN altitude.
```

Comment:
These checks are made before the MSA one. Also, the MSA simply crops any points above a given altitude, so it would not remove/create NaNs. The same criterion is not being applied to type 2 and 3, since these could be NaN.